### ISSUE TITLE
"pub finished with exit code 1" (and no error message)

### ISSUE THREAD

**Issue by tamoyal** [getsentry/sentry-dart-plugin#7]
We use sentry.io and the latest Flutter SDK. We're trying to get stack traces working on Android and it has been extremely difficult to get clarity on that process. We found this plugin and are still a bit confused how to use it.

If we have `autoUpload = true` in our `build.gradle` do we still need to run `flutter packages pub run sentry_dart_plugin` after we build? The comment here - https://docs.sentry.io/platforms/android/proguard/ - makes it seem like we wouldn't but then this plugin's README says that this command `flutter packages pub run sentry_dart_plugin` is a replacement for the sentry-cli manual stuff, NOT that autoupload setting. It would be good to clarify.

Also when running `flutter packages pub run sentry_dart_plugin` we get no error messages, just warnings, but we do get an exit code of `1` which idiomatically means error - "pub finished with exit code 1" ...so confused about that as well

**Comment by marandaneto** [comment]
@tamoyal thanks for reporting.

what type of stack traces are you meaning? Java/Kotlin stack traces, Dart or Native ones (C/C++)?

This is a Dart/Flutter plugin so there's nothing related to Proguard and the `autoUpload` flag, unless you are talking about the Java/Kotlin minified stack traces by Proguard/R8, which then requires the Sentry Android Gradle Plugin, can you please clarify, thanks

**Comment by tamoyal** [comment]
I think it's an issue with  the Java/Kotlin minified stack traces by Proguard/R8

We see a lot of stack traces like this:

```
_CastError: Null check operator used on a null value
  File "rEf", line 695, in _zja._dJd
  File "mFf", line 102, in Wka.bNd
```

Do I need to raise this issue somewhere else?

I brought it up here because the exit code of `1` with no error message is also a bit concerning

**Comment by marandaneto** [comment]
I dont think so, CastError is a Dart class https://api.dart.dev/stable/2.14.3/dart-core/CastError-class.html
can you share your build command? all you need is this plugin, not the Android Gradle Plugin, so you don't need to change your grade build file.

**Comment by tamoyal** [comment]
@marandaneto Ok so then something weird is definitely going on. Nothing special in our build command - `flutter build appbundle --flavor production --no-sound-null-safety`

**Comment by marandaneto** [comment]
@tamoyal why are you using this plugin then if you're not using the `--obfuscate` and `--split-debug-info` option? https://flutter.dev/docs/deployment/obfuscate

**Comment by tamoyal** [comment]
@marandaneto we set this in `gradle.properties` (see below) @marandaneto but I should mention, my experience is more on the iOS than Android side so something could be configured wrong. Given the stack traces are symbols, it looks like obfuscation is working, right?

The android/gradle.properties file:
```
org.gradle.jvmargs=-Xmx1536M
android.enableR8=true
android.useAndroidX=true
android.enableJetifier=true
android.bundle.enableUncompressedNativeLibs=false
extra-gen-snapshot-options=--obfuscate
```

**Comment by marandaneto** [comment]
@tamoyal as you see in the docs and Obfuscating-Dart-Code guideline, https://github.com/flutter/flutter/wiki/Obfuscating-Dart-Code the `--obfuscate` and `--split-debug-info` should be used together, otherwise, symbolication won't work.

please use both flags and this plugin to upload debug symbols and then it should work, let us know otherwise, thanks

**Comment by tamoyal** [comment]
@marandaneto Does the plugin expect the `--split-debug-info` output to be somewhere specific? I don't understand why there aren't step by step instructions in the README. It seems like it would be so easy to add and save everyone a lot of time trying to figure this out.

**Comment by marandaneto** [comment]
@tamoyal the folder just needs to be a child of the project's folder, the plugin finds symbols for all the subfolders, feel free to submit a PR to the readme, this is still an alpha version.

**Comment by marandaneto** [comment]
@tamoyal can you provide a minimal reproducible example? I can't reproduce it.

also, as stated in the readme now, iOS `--obfuscate` and `--split-debug-info` are not supported, so those files that are being skipped should not even be there, did you use those flags for iOS as well?

**Comment by marandaneto** [comment]
@tamoyal please use the latest version and give it a spin -> https://github.com/getsentry/sentry-dart-plugin/releases/tag/1.0.0-alpha.4
improved logging so at least we should not the error and output messages, also use `log_level: debug` in your config, might have something useful from sentry-cli output.
